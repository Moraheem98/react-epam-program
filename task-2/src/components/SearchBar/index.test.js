import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { fireEvent, render, screen } from '@testing-library/react';

const { SearchBar } = require('./index.js');
describe('Search component', () => {
	const initialState = { output: 10 };
	const mockStore = configureStore();
	let store;
	beforeAll(() => {
		store = mockStore(initialState);

		render(
			<Provider store={store}>
				<SearchBar />
			</Provider>,
		);
	});

	it('should have the right message in the dom', () => {
		const message = 'FIND YOUR MOVIE';
		expect(screen.getByText(message)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		let { asFragment } = render(
			<Provider store={store}>
				<SearchBar />
			</Provider>,
		);
		expect(asFragment(<SearchBar />)).toMatchSnapshot();
	});

	it('should simulate input', async () => {
		const { getByPlaceholderText } = render(
			<Provider store={store}>
				<SearchBar />
			</Provider>,
		);
		const inputID = getByPlaceholderText('Search any Movie...');
		fireEvent.change(inputID, { target: { value: 'ok' } });
	});
});
