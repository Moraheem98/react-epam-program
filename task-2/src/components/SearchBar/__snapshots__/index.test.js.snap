// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="searchContainer"
  >
    <h2>
      FIND YOUR MOVIE
    </h2>
    <input
      placeholder="Search any Movie..."
      type="text"
      value=""
    />
    <button
      class="appBtn"
      type="submit"
    >
      submit
    </button>
  </div>
</DocumentFragment>
`;
